components/eventing-controller/reconciler/subscription/reconciler_test.go

```diff
@@ -103,50 +103,158 @@ var _ = Describe("Subscription Reconciliation Tests", func() {
-			getK8sEvents(&subscriptionEvents, givenSubscription.Namespace).Should(reconcilertesting.HaveEvent(subscriptionDeletedEvent))
+			getK8sEvents(&subscriptionEvents, subscription1.Namespace).Should(reconcilertesting.HaveEvent(subscriptionDeletedEvent))
 
-			By("Sending valid webhook requests only")
-			validRequests, invalidRequests := countRequestsForWebhooks(reconcilertesting.EventType, reconcilertesting.EventTypeNotClean)
-			Expect(validRequests).Should(BeZero())
-			Expect(invalidRequests).Should(BeZero())
-		})
-	})
+			By("Ensuring the APIRule has 1 OwnerReference and 1 path")
+			getAPIRule(&apiRuleCreated, ctx).Should(And(
+				reconcilertesting.HaveNotEmptyHost(),
+				reconcilertesting.HaveNotEmptyAPIRule(),
+				reconcilertesting.HaveAPIRuleOwnersRefs(subscription2.UID),
+				reconcilertesting.HaveAPIRuleSpecRules(acceptableMethods, object.OAuthHandlerName, "/path2"),
+			))
 
-	// TODO: https://github.com/kyma-project/kyma/issues/9950
-	PWhen("Fixing an invalid Subscription (related to APIRule)", func() {
-		It("Should finally mark the subscription as ready", func() {})
-	})
+			By("Ensuring the deleted Subscription is removed as Owner from the APIRule")
+			getAPIRule(&apiRuleCreated, ctx).ShouldNot(And(
+				reconcilertesting.HaveAPIRuleOwnersRefs(subscription1.UID),
+				reconcilertesting.HaveAPIRuleSpecRules(acceptableMethods, object.OAuthHandlerName, "/path1"),
+			))
 
-	PWhen("Two Subscriptions using the same sink but different path", func() {
-		It("Should update the APIRule accordingly ", func() {})
-		// check APIRule has two owner refs and two paths
-		// delete one subscription
-		// check APIRule has one owner ref and one path
-		// delete last subscription
-		// check APIRule is gone
-	})
+			By("Deleting the second Subscription")
+			Expect(k8sClient.Delete(ctx, subscription2)).Should(BeNil())
+
+			By("Removing the Subscription")
+			getSubscription(subscription2, ctx).Should(reconcilertesting.IsAnEmptySubscription())
+
+			By("Emitting a k8s Subscription deleted event")
+			subscriptionDeletedEvent = v1.Event{
+				Reason:  string(eventingv1alpha1.ConditionReasonSubscriptionDeleted),
+				Message: "",
+				Type:    v1.EventTypeWarning,
+			}
+			getK8sEvents(&subscriptionEvents, subscription2.Namespace).Should(reconcilertesting.HaveEvent(subscriptionDeletedEvent))
+
+			By("Removing the APIRule")
+			Expect(apiRuleCreated.GetDeletionTimestamp).NotTo(BeNil())
 
-	PWhen("Creating two Subscriptions using different sinks", func() {
-		It("Should mark both Subscriptions as ready and create two different APIRules", func() {})
-		When("Unifying both Subscription sinks", func() {
-			It("Should mark both Subscriptions as ready and end up with only one APIRule", func() {})
+			By("Sending valid webhook requests only")
+			validRequests, invalidRequests := countRequestsForWebhooks(reconcilertesting.EventType, reconcilertesting.EventTypeNotClean)
+			Expect(validRequests).Should(reconcilertesting.BeGreaterThanOrEqual(2))
```

Comment:
Why do you expect at least 2 requests to BEB mock server? My take is 4:
- 2 creation reqs for sub1 and sub2
- 2 deletion reqs for sub1 and sub2

Reply:
The check only looked for requests with cleaned or not-clean event types

I have cleared this up by adding a check making sure there are no requests with invalid event types and then checking the minimum required amount of creation and deletion requests